### ISSUE TITLE
Reference specification for the `flutter` key in pubspec.yaml

### ISSUE THREAD

**Issue by jonasfj** [flutter/flutter#59480]
The format and semantics of features specified under the `flutter` key in `pubspec.yaml` often changes.

It might be wise to write up some reference docs for this. Notably to annotate what flutter versions supports which which semantics.

For example, prior to `1.20.0` all flutter plugins (`flutter.plugin != null`) was required to have an `ios/` folder. And prior to `1.12.0` the `flutter.plugin` schema looked very differently. Supporting such old plugin formats (correctly detecting platforms, and giving warnings) would be a lot easier if we knew at what flutter versions semantics changed, and what they were prior to that.

Maybe this is just a wishful nice to have, I don't know :D

**Comment by jonahwilliams** [comment]
I agree - in addition, it would be nice if all of the tools that need to parse or analyze this section could depend on a shared package to do so. That could be supported by extracting the current tool logic out and moving it to flutter/packages

**Comment by christopherfujino** [comment]
I'll just add that if we do this, we should rely on flutter stable version numbers. There's no programmatic way to know if stable version `1.20.0` is before or after dev release `1.20.0-1.2.pre`.
